packages/components/src/custom-select-control/index.js

```diff
@@ -75,6 +78,8 @@ export default function CustomSelectControl( {
 	/** @type {import('../select-control/types').SelectControlProps.size} */
 	size = 'default',
 	value: _selectedItem,
+	onMouseOver = noop,
```

Comment:
Nit: I don't think we need to default to `noop` in this case.

And if we ever need to incoke the function imperatively, we could always use the optional chaining syntax (`onMouseOver?.()`)

Reply:
The font size picker component throws a typescript error if I remove this. You happy to leave it there for now to stop this error, or is there another way to fix this without pulling unrelated typescript updates into this PR?

Reply:
Oh sure! That's a limitation with how TypeScript tries to infer `CustomSelectControl`'s prop types, since `CustomSelectControl` is not proving its own types.

The quick way to fix this, is to move the props destructuring inside the function (instead of in the function's arguments)

<details>
<summary>Like so</summary>

```diff
diff --git a/packages/components/src/custom-select-control/index.js b/packages/components/src/custom-select-control/index.js
index 45617b7928..4c8682f0c7 100644
--- a/packages/components/src/custom-select-control/index.js
+++ b/packages/components/src/custom-select-control/index.js
@@ -62,25 +62,25 @@ const stateReducer = (
 	}
 };
 
-const noop = () => {};
+export default function CustomSelectControl( props ) {
+	const {
+		/** Start opting into the larger default height that will become the default size in a future version. */
+		__next36pxDefaultSize = false,
+		/** Start opting into the unconstrained width that will become the default in a future version. */
+		__nextUnconstrainedWidth = false,
+		className,
+		hideLabelFromVision,
+		label,
+		describedBy,
+		options: items,
+		onChange: onSelectedItemChange,
+		/** @type {import('../select-control/types').SelectControlProps.size} */
+		size = 'default',
+		value: _selectedItem,
+		onMouseOver,
+		onMouseOut,
+	} = props;
 
-export default function CustomSelectControl( {
-	/** Start opting into the larger default height that will become the default size in a future version. */
-	__next36pxDefaultSize = false,
-	/** Start opting into the unconstrained width that will become the default in a future version. */
-	__nextUnconstrainedWidth = false,
-	className,
-	hideLabelFromVision,
-	label,
-	describedBy,
-	options: items,
-	onChange: onSelectedItemChange,
-	/** @type {import('../select-control/types').SelectControlProps.size} */
-	size = 'default',
-	value: _selectedItem,
-	onMouseOver = noop,
-	onMouseOut = noop,
-} ) {
 	const {
 		getLabelProps,
 		getToggleButtonProps,
```

</details>

That should fix the TypeScript issue, removing the need for the `noop` without unrelated typescript updates 🤞 

For more TypeScript-related shenanigans like the one above, you can also take a look at our [TypeScript migration guide](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/CONTRIBUTING.md#refactoring-a-component-to-typescript)

Reply:
Ah, perfect, that fixed it, thanks!